kbchat/kvstore_test.go

```diff
@@ -0,0 +1,102 @@
+package kbchat
+
+import (
+	"testing"
+
+	"github.com/keybase/go-keybase-chat-bot/kbchat/types/keybase1"
+	"github.com/stretchr/testify/require"
+)
+
+func contains(a []string, x string) bool {
+	for _, n := range a {
+		if x == n {
+			return true
+		}
+	}
+	return false
+}
+
+func containsKey(a []keybase1.KVListEntryKey, x string) bool {
+	for _, n := range a {
+		if x == n.EntryKey {
+			return true
+		}
+	}
+	return false
+}
+
+// test doesn't make assumption about prior state
+func TestKVStore(t *testing.T) {
+	var err error
+
+	alice, dir := testBotSetup(t, "alice")
+	channel := getTeamChatChannel(t, "acme")
+	team := channel.Name
+	defer testBotTeardown(t, alice, dir)
+
+	namespace := "_test_namespace1"
+	key := "_test_key1"
+
+	// put with default revision
+	put, err := alice.PutEntry(team, namespace, key, "value1")
+	require.NoError(t, err)
+	require.True(t, put.Revision > 0)
+	rev := put.Revision
+
+	expectedRevision := rev + 1
+
+	// fail put (wrong revision)
+	_, err = alice.PutEntryWithRevision(team, namespace, key, "value2", expectedRevision-1)
+	require.Error(t, err)
+	require.Equal(t, RevisionErrorCode, err.(Error).Code)
```

Comment:
`err.(Error)` made me do a double take. it's weird that we wrapped `error` with `Error`. looks like it was in the library before. not a blocker. i just don't love it.

Reply:
maybe i'm missing something here, but isn't an Error struct necessary for unmarshaling the cli json response?